subprojects/core/src/main/java/org/gradle/api/internal/file/copy/DefaultFileCopyDetails.java

```diff
@@ -149,22 +160,22 @@ public RelativePath getRelativePath() {
     }
 
     @Override
-    public int getMode() {
-        if (mode != null) {
-            return mode;
+    public Provider<ImmutableFileAccessPermissions> getImmutablePermissions() {
+        if (permissions != null) {
+            permissions.finalizeValue();
+            return Providers.of(new DefaultImmutableFileAccessPermissions(permissions.flatMap(ImmutableFileAccessPermissions::toUnixNumeric).get()));
```

Comment:
We should strive at not using `Providers.of()` when possible in order to not break provider chains.

It looks like this could be replaced with:
```java
permissions.flatMap(ImmutableFileAccessPermissions::toUnixNumeric).map(DefaultImmutableFileAccessPermissions::new)
```